docs/design/simplified_federation_deployment.md

```diff
@@ -0,0 +1,387 @@
+
+**1. Remove the cluster resource from federation**
+
+* Remove the cluster API resource corresponding to `<cluster-context-
+name>` from federation.
+
+**2. Remove the credentials secret**
+
+* Remove the kubeconfig secret that we created during the registration
+  process from the bootstrap cluster.
+
+**Note:** `--bootstrap-cluster` flag is only required because federation
+still doesn't support selecting specific clusters in federated secrets.
+It copies them to all the underlying clusters. It is unnecessary to copy
+the credentials in clusters other than the bootstrap cluster. And since
+these are secrets that include cluster credentials, we want to err on
+the side of being too conservative. This flag can be deprecated once
+federated secrets gain cluster selection support.
+
+## Test Plan
+
+#### Federation Control Plane Bootstrap
+
+End-to-end testing of `kubeadm` functionality is somewhat tricky because
+`kubeadm init` and hence `kubeadm init federation` prints out a token to
+STDOUT that must be used in all the `kubeadm join` invocations on the
+non-master Kubernetes nodes. Given that, here is an overview of how we
+plan to test the federation bootstrap functionality.
+
+**1. kubernetes-anywhere for resource provisioning**
+
+* We use
+  [kubernetes-anywhere](https://github.com/kubernetes/kubernetes-anywhere)
+  to provision cloud resources, i.e. VMs, Networking etc.
+
+**2. Provide a customized VM startup script**
+
+* We provide a customized `configure-vm.sh` script to run during VM
+  startup.
+* On the master VM, this script runs `kubeadm init federation` which
+  prints out a token.
+* The token is written to an object store (GCS on GCP and S3 on AWS).
+* On the non-master nodes, `configure-vm.sh` script polls for the token
+  object in the pre-configured storage bucket.
+* Once that appears, it uses the token to run `kubeadm join <token>`.
+
+**3. Verify that the federation control plane is up and running**
+
+* We verify that the federation control plane pods are up and are in
+  "Running" state.
+* We verify that the request to a discovery endpoint on the federation
+  API server returns a valid response.
+
+#### Cluster Registration/Deregistration
+
+This should be as simple as running `kubectl register <cluster-context>`
+and verifying that the registered cluster reports "Ready" status.
+
+Running `kubectl get clusters` should not list the unregistered cluster.
+
+
+## Timeline
+
+Both the federation control plane deployment functionality via `kubeadm`
+and the cluster registration/deregistration functionality via `kubectl`
+are planned to be implemented in the v1.5 time frame and are considered
+a P0 feature for v1.5.
+
+
+## Maturity Level
+
+Both functionalities will be labeled "alpha" in v1.5.
+
+
+<!-- BEGIN MUNGE: GENERATED_ANALYTICS -->
+[![Analytics](https://kubernetes-site.appspot.com/UA-36037335-10/GitHub/docs/design/simplified_federation_deployment.md?pixel)]()
+<!-- END MUNGE: GENERATED_ANALYTICS -->
```

Comment:
I agree in general that register / join are much better suited for kubeadm.  kubectl needs to have a coherent purpose, and some actions are too specific to be part of kubect